docs/deploy-lab.md

```diff
@@ -0,0 +1,355 @@
+# Enabling blue/green deployment with canary
+
+Automating deployment is critical to the success of most projects. However, in today's world, it's not enough to just deploy your code. You also need to ensure downtime is limited (or eliminated), and you can quickly rollback in the event of a failure. One common approach to this is to use a [blue/green deployment strategy](https://martinfowler.com/bliki/BlueGreenDeployment.html). This strategy involves deploying your code to a new environment, and then slowly shifting traffic from the old environment to the new environment. This allows you to test your new code in production, and quickly rollback if there are any issues.
+
+In this tutorial, you'll explore how to use [Azure Container Apps](https://learn.microsoft.com/en-us/azure/container-apps) to deploy a new version of your application, and then use Azure Traffic Manager to shift traffic from the old environment to the new environment. You'll start by creating and configuring the necessary Azure resources. You'll then configure a workflow in GitHub to deploy your application with a [canary](https://www.opsmx.com/blog/what-is-canary-deployment/) to automate rollback. Finally, you'll test the deployment and rollback process by commiting a change and monitoring the workflow.
+
+## Create an Azure account and install resources
+
+This workshop uses Azure Container Apps as the cloud-based host for the NGINX application. You'll need to create an Azure account and install the Azure CLI to complete this tutorial.
```

Comment:
I'd suggest we characterize it as "an NGINX webserver playing the role of the application" since NGINX can be configured in many ways.  Web server is a common configuration but it's not generally referred to as an application.

Reply:
Will update!! As a side note, one little GitHub tip is if you hit the little +/- button you can make suggestions inline which can quickly be committed into the branch right here in the conversation.

Reply:
Oh yeah I always forget about that but someone did it for me recently and I appreciated it.  Adding to my tool belt..

Reply:
No worries! I've honestly been surprised at the number of people even inside GitHub who haven't seen the feature. It's one of my favorites

Reply:
I love that feature, although I sometimes get carried away and write myself all the PR changes I was going to ask 😂